lib/ontologies_linked_data/models/agents/agent.rb

```
@@ -81,6 +81,72 @@ def keywords(force_update: false)
+        q = Goo.sparql_query_client.select(:id, :agent).distinct.from(LinkedData::Models::OntologySubmission.uri_type).where([:id, :property, :agent])
+        q = q.filter(OntologySubmission.agents_attr_uris.map{|attr| "?property = <#{attr}>"}.join(' || '))
+        q = q.values(:id,  *agent.usages.keys.map { |uri| RDF::URI(uri.to_s)})
+        relatedAgentsIds = q.each_solution.group_by{|x| x[:agent].to_s}
+                            .reject { |agent_id, _| agent_id == agent.id.to_s }
+                            .transform_values { |solutions| solutions.map { |s| s[:id] } }
+        # map the previously fetched usages
+        relatedAgents = self.fetch_agents_data(relatedAgentsIds.keys).each { |agent| agent.usages = relatedAgentsIds[agent.id.to_s].map(&:to_s).uniq }
+        
+        agent.instance_variable_set("@relatedAgents", relatedAgents)
+        agent.loaded_attributes.add(:relatedAgents)
+      end     
```

RuboCop (1.75.5)

[convention] Assignment Branch Condition size for load_related_agents is too high. [<13, 38, 9> 41.16/23]

(Metrics/AbcSize)